crates/ruff/src/pyproject_toml.rs

```diff
@@ -0,0 +1,36 @@
+use anyhow::Result;
+use pyproject_toml::PyProjectToml;
+use ruff_text_size::{TextRange, TextSize};
+
+use ruff_diagnostics::Diagnostic;
+use ruff_python_ast::source_code::SourceFile;
+
+use crate::message::Message;
+use crate::rules::ruff::rules::InvalidPyprojectToml;
+
+pub fn lint_pyproject_toml(source_file: SourceFile) -> Result<Vec<Message>> {
+    let err = match toml::from_str::<PyProjectToml>(source_file.source_text()) {
+        Ok(_) => return Ok(Vec::default()),
+        Err(err) => err,
+    };
+
+    let range = match err.span() {
+        // This is bad but sometimes toml and/or serde just don't give us spans
+        None => TextRange::default(),
```

Comment:
Can we follow up on this with an issue to make `TextRange` and `SourceFile` on `Diagnostic` optional and then link to it here in a TODO?

Reply:
done